specs/altair/beacon-chain.md

```diff
@@ -455,22 +470,22 @@ def process_attestation(state: BeaconState, attestation: Attestation) -> None:
         justified_checkpoint = state.previous_justified_checkpoint
 
     # Matching roots
-    is_matching_head = data.beacon_block_root == get_block_root_at_slot(state, data.slot)
     is_matching_source = data.source == justified_checkpoint
-    is_matching_target = data.target.root == get_block_root(state, data.target.epoch)
+    is_matching_target = is_matching_source and data.target.root == get_block_root(state, data.target.epoch)
+    is_matching_head = is_matching_target and data.beacon_block_root == get_block_root_at_slot(state, data.slot)
```

Comment:
The condition changed from just a head block-root check, to a is-target + head block root check. And the if statements below changed to then not do the target check again. I think we were testing good head votes with bad target votes, but if not, I'll try add this after the specs call.

Reply:
This is not the case. See here:

https://github.com/ethereum/eth2.0-specs/pull/2374/files#diff-76363249532da52344c30ccc6864859e2a74ad83130d0d640033cbbe4ff0a5beL468

We used to write `is_matching_head and is_matching_target` whereas this changes `is_matching_head` to include `is_matching_target` to remove the footgun of using head alone somewhere by accident

Reply:
EDIT
I misread your statement. You are correct in your assessment and we almost certainly have such tests

Reply:
> And the if statements below changed to then not do the target check again.

That's what I said, "to not check the target *again*", after it being checked as part of `is_matching_head`. Anyway, it looks consistent, but we need to check testing.

Reply:
Confirmed we have tests for it. Bad target while a valid head vote is processed, with multiple cases for inclusion delay too: https://github.com/ethereum/eth2.0-specs/blob/0c94be420405380f104347e643500f4613ab8f9c/tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py#L498